bika/lims/browser/sample/view.py

```diff
@@ -514,3 +522,176 @@ def _schedule_sampling_permissions(self):
+            'type': 'select',
+            'voc': self._getSampleTypesVoc(),
+        }, {
+            'name': 'case',
+            'label': _('Cases'),
+            'type': 'autocomplete_text',
+            'voc': json.dumps(self._getCasesVoc()),
+        }, {
+            'name': 'date_received',
+            'label': _('Date received'),
+            'type': 'date_range',
+        },
+        ]
+        return fields_dict
+
+    def _getSampleConditionsVoc(self):
+        """
+        Returns a DisplayList object with sample condtions.
+        """
+        cons = self.context.bika_setup.\
+            bika_sampleconditions.listFolderContents()
+        return DisplayList(
+            [(element.UID(), element.Title()) for element in cons])
+
+    def _getPrintStatesVoc(self):
+        """
+        Returns a DisplayList object with print states.
+        """
+        return DisplayList([
+            ('0', _('Never printed')),
+            ('1', _('Printed after last publish')),
+            ('2', _('Printed but republished afterwards')),
+            ])
+
+    def _getSampleTypesVoc(self):
+        """
+        Returns a DisplayList object with sample types.
+        """
+        types = self.context.bika_setup.bika_sampletypes.listFolderContents()
+        return DisplayList(
+            [(element.UID(), element.Title()) for element in types])
+
+    def _getCasesVoc(self):
+        """
+        Returns a list object with active cases ids.
+        """
+        catalog = getToolByName(self.context, "portal_catalog")
+        brains = catalog({
+            'portal_type': 'Batch',
+            'review_state': 'open',
+        })
+        return [brain.id for brain in brains]
+
+    def get_filter_bar_queryaddition(self):
+        """
+        This function gets the values from the filter bar inputs in order to
+        create a query accordingly.
+        Only returns the once that can be added to contentFilter dictionary.
+        in this case, the catalog is bika_catalog
+        In this case the keys with index representation are:
+        - date_received - getDateReceived
+        - date_received - BatchUID
+        :return: a dictionary to be added to contentFilter.
+        """
+        query_dict = {}
+        filter_dict = self.get_filter_bar_dict()
+        # Date received filter
+        if filter_dict.get('date_received_0', '') or\
+                filter_dict.get('date_received_1', ''):
+            date_0 = filter_dict.get('date_received_0') \
+                if filter_dict.get('date_received_0', '')\
+                else '1900-01-01'
+            date_1 = filter_dict.get('date_received_1')\
+                if filter_dict.get('date_received_1', '')\
+                else datetime.strftime(date.today(), "%Y-%m-%d")
+            date_range_query = {
+                'query':
+                (date_0 + ' 00:00', date_1 + ' 23:59'), 'range': 'min:max'}
+            query_dict['getDateReceived'] = date_range_query
```

Comment:
I've seen this logic to get the date range repeated many times. Maybe better if we move this logic to BaseBikaListingFilterBar or create a function in utils